### ISSUE TITLE
VS Code incorrectly detects rebase in progress

### ISSUE THREAD

**Issue by schroding3rscat** [Microsoft/vscode#57299]
Issue Type: <b>Bug</b>

After working with a code and before committing anything to a repo, I stashed all changes and started rebase from SourceTree. VS Code showed me conflicts properly. I did all merge operations in Visual Studio 2017, committed and pushed all changes. There was a bunch of other git operations later. I closed and opened VS Code, restarted PC, did some coding, committed and pushed something. But still, I see a warning that I cannot change commit message in the middle of a rebase:
<img width="192" alt="screenshot_4" src="https://user-images.githubusercontent.com/3431598/44689443-3e15d280-aa60-11e8-816f-ab506376fbff.png">

If I try to commit anything from VS Code, I see error popup:
<img width="281" alt="screenshot_5" src="https://user-images.githubusercontent.com/3431598/44689444-3e15d280-aa60-11e8-946a-60d4c4585a8a.png">

When I press commit button, I see new lines added to git output in VS Code:
> git rebase --continue
No rebase in progress?
> git config --get-all user.name
> git config --get-all user.email

All other git clients working fine and I can do all git operations. Git syncing and branch switching don't help.
`git status` output:
```
λ git status
On branch feature/dev-trials-support
Your branch is up to date with 'origin/feature/dev-trials-support'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   docker-compose.yml
```

I didn't manage to get rid of this, so I assume there's a bug with the git module.

VS Code version: Code 1.26.1 (493869ee8e8a846b0855873886fc79d480d342de, 2018-08-16T18:38:57.434Z)
OS version: Windows_NT x64 10.0.17134

<details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
RPMSpec|1do|0.1.0
unique-lines|bib|1.0.0
vscode-raml|blz|3.0.1
dart-code|Dar|2.17.1
flutter|Dar|2.17.1
EditorConfig|Edi|0.12.4
LogFileHighlighter|emi|2.4.0
vsc-material-theme|Equ|2.3.0
vscode-guid|hea|1.3.1
rest-client|hum|0.19.1
ecdc|mit|0.12.0
python|ms-|2018.7.1
csharp|ms-|1.15.2
Go|ms-|0.6.88
vscode-yaml|red|0.0.15
vscode-icons|rob|7.25.0
vscode-todo-highlight|way|1.0.4
change-case|wma|1.0.0

(4 theme extensions excluded)

</details>

**Comment by schroding3rscat** [comment]
> How did you commit. If you are in a middle of a rebase, you should've used git rebase --continue

I did all the work in SourceTree as usual. SourceTree, Visual Studio 2017 and `git status` don't show any rebase process now. Why does VS Code do?
I do have `.git/REBASE_HEAD`. But before deleting it I want to know why VS Code behaves like this.

**Comment by joaomoreno** [comment]
VS Code checks whether that file exists to determined if there's a rebase in flow. Apparently that check is insufficient and we should also check for other signs of rebase.

**Comment by joaomoreno** [comment]
Can you show me the full output of `git status`?

**Comment by joaomoreno** [comment]
Here's some reference for fixing this: https://stackoverflow.com/questions/3921409/how-to-know-if-there-is-a-git-rebase-in-progress

**Comment by schroding3rscat** [comment]
> Can you show me the full output of git status?

I'm afraid I can't. I checked the git section in vscode and it seems that problem disappeared now. Today I did a rebase in SourceTree again and switched branches. I think this helped and `REBASE_HEAD` was removed.
But I added `git status` output to the bug description when I created it yesterday. I run this command when I had a problem.

**Comment by jeanbenitez** [comment]
Hi! I had the same issue, I solved removing .git/REBASE_HEAD:
`rm -rf .git/REBASE_HEAD`

**Comment by joaomoreno** [comment]
Fixed by #57651
